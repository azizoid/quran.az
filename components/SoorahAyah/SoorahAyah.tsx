import Link from 'next/link'
import { FaExternalLinkAlt } from 'react-icons/fa'
import { DisplayData } from '@/lib/types'
import { Sajda } from '@/ui'

export type SoorahAyahProps = {
  data: DisplayData
  sajda?: number[]
}

export const SoorahAyah = ({ data, sajda }: SoorahAyahProps): JSX.Element => (
  <li className="soorah-list-item">
    <div className="flex flex-row">
      <span className="badge">
        {data.ayah}
        {sajda?.includes(data.ayah) && <Sajda />}
      </span>
      <span>{data.content}</span>
    </div>
    <Link
      href={`/${data.soorah}/${data.ayah}?t=${data.translator}`}
      className="read-ayah">

      <FaExternalLinkAlt />

    </Link>
  </li>
)
