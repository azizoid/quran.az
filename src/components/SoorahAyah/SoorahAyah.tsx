import Link from 'next/link'
import { FaExternalLinkAlt } from 'react-icons/fa'

import { DisplayData } from 'src/lib/types'
import { Sajda } from 'src/ui'

export type SoorahAyahProps = {
  data: DisplayData
  sajda?: number[]
}

export const SoorahAyah = ({ data, sajda }: SoorahAyahProps): JSX.Element => (
  <li className="soorah-list-item">
    <div className="flex flex-row">
      <span className="badge">
        {data.ayah}
        {sajda?.includes(data.ayah) && <Sajda />}
      </span>
      <span>{data.content}</span>
    </div>
    <Link href={`/${data.soorah}/${data.ayah}?t=${data.translator}`} className="read-ayah">
      <FaExternalLinkAlt />
    </Link>
  </li>
)
