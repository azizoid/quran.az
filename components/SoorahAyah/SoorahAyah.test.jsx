import React from 'react'
import { render } from '@testing-library/react'
import { SoorahAyah } from './SoorahAyah'

test('SoorahAyah Snapshot', () => {
  const { container } = render(
    <SoorahAyah
      data={{
        soorah: 96,
        ayah: 1,
        content: 'Yaradan Rəbbinin adı ilə oxu!',
        translator: 4,
      }}
    />
  )

  expect(container).toMatchSnapshot()
})
