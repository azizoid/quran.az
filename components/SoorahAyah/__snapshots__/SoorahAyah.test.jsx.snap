// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoorahAyah Snapshot 1`] = `
<div>
  <li
    class="soorah-list-item"
  >
    <div
      class="flex flex-row"
    >
      <span
        class="badge"
      >
        1
      </span>
      <span>
        Yaradan Rəbbinin adı ilə oxu!
      </span>
    </div>
    <a
      class="read-ayah"
      href="/96/1?t=4"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </li>
</div>
`;
